File: examples/python/evm/smart_account.list_token_balances.py
# Usage: uv run python evm/smart_account.list_token_balances.py

import asyncio
from cdp import CdpClient
from web3 import Web3
from dotenv import load_dotenv

load_dotenv()


async def main():
    async with CdpClient() as cdp:
        account = await cdp.evm.get_or_create_account(name="MyAccount")
        smart_account = await cdp.evm.create_smart_account(owner=account)

        transaction_hash = await cdp.evm.request_faucet(
            address=smart_account.address,
            network="base-sepolia",
            token="eth",
        )
        w3 = Web3(Web3.HTTPProvider("https://sepolia.base.org"))
        w3.eth.wait_for_transaction_receipt(transaction_hash)

        balances = await smart_account.list_token_balances(
            network="base-sepolia",
            page_size=10,
        )

        for balance in balances.balances:
            print(f"Token contract address: {balance.token.contract_address}")
            print(f"Token balance: {balance.amount.amount}")


asyncio.run(main())

File: examples/typescript/evm/smartAccount.listTokenBalances.ts
// Usage: pnpm tsx evm/smartAccount.listTokenBalances.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import { createPublicClient, http } from "viem";
import { baseSepolia } from "viem/chains";
import "dotenv/config";

const cdp = new CdpClient();

const account = await cdp.evm.getOrCreateAccount({ name: "MyAccount" });
const smartAccount = await cdp.evm.createSmartAccount({
  owner: account,
});

const { transactionHash } = await cdp.evm.requestFaucet({
  address: smartAccount.address,
  network: "base-sepolia",
  token: "eth",
});

await createPublicClient({
  transport: http(),
  chain: baseSepolia,
}).waitForTransactionReceipt({
  hash: transactionHash,
});

const result = await smartAccount.listTokenBalances({
  network: "base-sepolia",
});

for (const balance of result.balances) {
  console.log("Token contract address:", balance.token.contractAddress);
  console.log("Token balance:", balance.amount.amount);
}
